controllers/helmchartinstall_controller.go

```diff
@@ -77,61 +126,255 @@ func (r *HelmChartInstallReconciler) render(ctx context.Context, claim *helmv1al
 	- Watch for new types being provided by stacks
 	- When there is a new type provided, watch for instances being created
 	- When an instance is created, feed it into the renderer
+
+	TODO This could use a bit of refactoring so it flows more nicely
 	*/
 
 	// configuration is a typed object
-	configuration, err := r.getStackConfiguration()
+	configuration, err := r.getStackConfiguration(ctx, claim)
+	// TODO check for errors
+
+	targetResourceBehavior, err := r.getBehavior(ctx, claim, configuration)
+
+	if targetResourceBehavior == nil {
+		// TODO error condition with a real error returned
+		r.Log.V(0).Info("Couldn't find a configured behavior!",
+			"claim", claim,
+			"configuration", configuration,
+		)
+		return err
```

Comment:
I think your TODO may already capture this, but it feels a little strange here that you return err when `targetResourceBehaviour == nil` rather than when `err != nil`.

Reply:
I've found myself handling `err != nil { return err }` and `thing == nil { return fmt.Errorf('thing was nil')}`.  

Whenever I find myself doing this, I rethink the function and why I would ever return `thing=nil` without an error.  Almost always an oversight.